src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -10,7 +10,7 @@
 public class Tag {
 
     public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
-    public static final String VALIDATION_REGEX = ".+";
+    public static final String VALIDATION_REGEX = "^[a-zA-Z0-9]+$";
```

Comment:
Corrected from #169

Reply:
Modified in latest commit to allow underscores and dashes

![image](https://user-images.githubusercontent.com/37281349/48309507-c9f6bf80-e5b6-11e8-960b-d0466caa3153.png)